### PR TITLE
Alter table

### PR DESCRIPTION
**Pull request by kimitaWanjohi**
- Run some update queries and transactions

here are some screenshots: 

- > inside a transaction update the animals table by setting the species column to unspecified 
- > Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- > Inside a transaction delete all records in the animals table, then roll back the transaction.
 

- > Delete all animals born after Jan 1st, 2022.
- > Create a savepoint for the transaction.
- > Update all animals' weight to be their weight multiplied by -1.
- > Rollback to the savepoint
- > Update all animals' weights that are negative to be their weight multiplied by -1. 

# QUERIES 
- - > How many animals are there? 
- - > How many animals have never tried to escape? 
- - > What is the average weight of animals? 
- - > Who escapes the most, neutered or not neutered animals? 
- - > What is the minimum and maximum weight of each type of animal? 
- > What is the average number of escape attempts per animal type of those born between 1990 and 2000? 

# fixes 

- > Fix the species issue 

### fixed queries 

- > What is the minimum and maximum weight of each type of animal? 
- > What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -8,4 +8,25 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+VALUES ('Angemon', '2005-06-12', 1, true, -45, 'Angel');
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Boarmon', '2005-06-07', 7, true, 20.4, 'Mammal');
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Blossom', '1998-10-13', 3, true, 17, 'Flower');
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Ditto', '2022-05-14', 4, true, 22, 'Normal');
```
Comment (Owusu-Desmond) [inline data.sql]: - [x] Please you are inserting values into the species column which wasn't required in the project requirement so I think you should not insert any values to the species column when inserting these data (The species column should be left null) :pray:

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,52 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
```
Comment (Owusu-Desmond) [inline queries.sql]: - [x] Please you missing the queries for these two requirement :point_down: so make sure the queries are present in this file :pray: 
    
    - Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction.
    - Inside a transaction:
      - Update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
      - Update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
      - Verify that changes were made.
      - Commit the transaction.
      - Verify that changes persist after commit.

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @kimitaWanjohi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kimitaWanjohi** [comment]
@Owusu-Desmond  I've already done that in the fixes in the pr description

**Comment by Owusu-Desmond** [comment]
@kimitaWanjohi Yes, I can see the screenshot in the PR description but make sure that the queries are present in the queries.sql file

File: data.sql (hunk 1 above)
Comment (Qoosim) [reply data.sql]: > * [x]  Please you are inserting values into the species column which wasn't required in the project requirement so I think you should not insert any values to the species column when inserting these data (The species column should be left null) pray

- [REQUIRED] Kindly attend to this required change from the previous reviewer. It is yet to be attended to. Thanks

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @kimitaWanjohi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  Fair README file. :heavy_check_mark: 
-  Correct . :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kimitaWanjohi** [comment]
@Owusu-Desmond I honestly don't understand what you are requesting mate, but I have added the the update to data.sql to change the species name

**Review by TadesseDev — APPROVED** [review]
### Hi @kimitaWanjohi 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewers. ✅


## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**